tokio/src/stream/mod.rs

```diff
@@ -680,6 +686,69 @@ pub trait StreamExt: Stream {
     {
         Collect::new(self)
     }
+
+    /// Applies a per-item timeout to the passed stream.
+    ///
+    /// `timeout()` takes a `Duration` that represents the maximum amount of
+    /// time each element of the stream has to complete before timing out.
+    ///
+    /// If the wrapped stream yields a value before the deadline is reached, the
+    /// value is returned. Otherwise, an error is returned. The caller may decide
+    /// to continue consuming the stream and will eventually get the next source
+    /// stream value once it becomes available.
+    ///
+    /// # Notes
+    ///
+    /// This function consumes the stream passed into it and returns a
+    /// wrapped version of it.
+    ///
+    /// The wrapped stream will continue to be polled even if one or more items
+    /// time out.
```

Comment:
I feel like this kind of implies that the stream will continue to be polled _in the background_, which isn't the case. What do you think about
> Polling the returned stream will continue to poll the inner stream even if one or more items time out.

Reply:
Done!